tests/5.1/teams/test_target_get_max_teams.c

```diff
@@ -0,0 +1,40 @@
+//===--------------------- test_target_get_max_teams.c ----------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test uses the omp_get_max_teams routine to check what the max teams 
+// capacity is for this device. It should return the total amount of teams
+// that could be allocated to a specific teams region
+//
+//===------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int main() {
+
+	OMPVV_TEST_OFFLOADING;
+
+	int errors = 0; 
+	int MAX_TEAMS = omp_get_max_teams();
+	int num_teams = MAX_TEAMS + 1; //a value that is not possible
+
+	#pragma omp target teams map(tofrom: num_teams)
+	{
+		if (omp_get_team_num() == 0) {
+			num_teams = omp_get_num_teams();
+		}
+	}               
+
+        OMPVV_ERROR_IF(MAX_TEAMS > 0 && num_teams > MAX_TEAMS, "Number of teams reported exceeded max number of teams (max no. > 0)");
+	OMPVV_ERROR_IF(num_teams < 0, "Number of teams is negative");
```

Comment:
How about checking slightly stricter:
```suggestion
	OMPVV_ERROR_IF(num_teams <= 0, "Number of teams is not positive");
```
as OpenMP requires that at least 1 team is created. (→ num_teams clause description in 'teams construct' section.)

_In any case, this condition is unlikely to fail. It can semantically only fail if omp_get_max_teams() returns <= -1  (suggestion above) or < -1 (current code) or if omp_get_num_teams(); returns <= 0 / < 0 while at the same time the region is executed with omp_get_team_num() == 0_
* * *
I assume it will work just fine with most implementations, i.e. as long as omp_get_max_teams() returns the current ICV value (which is known to be zero).

However, technically, omp_get_max_teams() relates to the host – while omp_get_num_teams() is executed on the device. — I see no restriction in OpenMP that host and device must have the same implementation-defined maximal number of teams.

Reply:
Agree. Will split it up into host and device checks.